### PR TITLE
Re-enable the Systemd remote support

### PR DESCRIPTION
**Pull request by nlewo**
###### Motivation for this change

Since ae896e0, Systemd is no longer compiled with the http gateway because of an incompatibility between Systemd and libmicrohttpd. However, because of some updates, it is now building fine!

I also enable this feature is the systemd test to ensure the option services.journald.enableHttpGateway is working as excepted.

@flokli should i rebase these patches onto `staging-20.09`? 
Note I didn't run the systemd tests yet since a lot of things need to be rebuilt:/

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by flokli** [comment]
@nlewo with https://github.com/NixOS/nixpkgs/pull/101886 merged, can you rebase this on top of this?

File: pkgs/os-specific/linux/systemd/default.nix [hunk 1]
```diff
@@ -112,7 +112,6 @@ in stdenv.mkDerivation {
     "-Dhostnamed=true"
     "-Dnetworkd=true"
     "-Dportabled=false"
-    "-Dremote=false"
```
Comment (flokli) [inline pkgs/os-specific/linux/systemd/default.nix:115]: This should be set explicitly to true, not implicitly use defaults.

File: nixos/tests/systemd.nix [hunk 2]
```diff
@@ -24,6 +24,10 @@ import ./make-test-python.nix ({ pkgs, ... }: {
     services.journald.extraConfig = "Storage=volatile";
     test-support.displayManager.auto.user = "alice";
 
+    # To ensure systemd has been compiled with the
+    # systemd-journal-gatewayd support.
+    services.journald.enableHttpGateway = true;
```
Comment (flokli) [inline nixos/tests/systemd.nix:29]: Can we test for some service being enabled and running, and not just enable this option?

**Review by flokli — CHANGES_REQUESTED** [review]

**Comment by flokli** [comment]
We should also first fix the default of `withRemote` in staging.

File: nixos/tests/systemd.nix (hunk 2 above)
Comment (nlewo) [reply nixos/tests/systemd.nix:27]: I added a http request to ensure the journal-gatewayd service is up and running.

File: pkgs/os-specific/linux/systemd/default.nix (hunk 1 above)
Comment (nlewo) [reply pkgs/os-specific/linux/systemd/default.nix:115]: Done.

File: nixos/tests/systemd.nix (hunk 2 above)
Comment (flokli) [reply nixos/tests/systemd.nix:27]: Thanks!

**Comment by flokli** [comment]
This should target `release-20.09`, not `nixos-20.09`.

It kinda backports the commtis from https://github.com/NixOS/nixpkgs/pull/102448 - even though it's not a real cherry-pick, but a logical backport, can we add the `(cherry picked from commit xxx) to the commit mesages, and align the commit messages themselves, to better cross-link this?

**Comment by nlewo** [comment]
@flokli ok, i rewrote the history and based the branch onto `release-20.09`.

Note the test is still building on my laptop...

**Comment by flokli** [comment]
Updated this to target `staging-20.09`.

**Comment by flokli** [comment]
Had to fix some indentation on the test, otherwise succeeded :+1:

**Comment by nlewo** [comment]
@flokli Cool, thank you!

**Pull request merged**